OutOfSchool/OutOfSchool.WebApi/Controllers/AddressController.cs

```diff
@@ -0,0 +1,142 @@
+    public class AddressController : ControllerBase
+    {
+        private readonly ILogger<AddressController> logger;
+        private readonly IAddressService addressService;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AddressController"/> class.
+        /// </summary>
+        /// <param name="logger">Logging instance.</param>
+        /// <param name="addressService">Service for Address model.</param>
+        public AddressController(ILogger<AddressController> logger, IAddressService addressService)
+        {
+            this.logger = logger;
+            this.addressService = addressService;
+        }
+
+        /// <summary>
+        /// Get all addresses from the database.
+        /// </summary>
+        /// <returns>List of all addresses.</returns>
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        public async Task<IActionResult> GetAddresses()
+        {
+            return Ok(await addressService.GetAll().ConfigureAwait(false));
+        }
+
+        /// <summary>
+        /// Get address by it's key.
+        /// </summary>
+        /// <param name="id">The key in the database.</param>
+        /// <returns>Address element with some id.</returns>
+        [HttpGet("{id}")]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        public async Task<IActionResult> GetAddressById(long id)
+        {
+            if (id == 0)
+            {
+                return BadRequest("Id cannot be 0.");
+            }
+
+            return Ok(await addressService.GetById(id).ConfigureAwait(false));
+        }
+
+        /// <summary>
+        /// Create new address.
+        /// </summary>
+        /// <param name="addressDto">Element which must be added.</param>
+        /// <returns>A <see cref="Task{TResult}"/> representing the result of the asynchronous operation.</returns>
+        [Authorize(Roles = "provider,admin")]
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status201Created)]
+        [ProducesResponseType(StatusCodes.Status401Unauthorized)]
+        public async Task<IActionResult> Create(AddressDto addressDto)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            if (addressDto == null)
+            {
+                return BadRequest("Address is null.");
+            }
+
+            try
+            {
+                AddressDto address = await addressService.Create(addressDto).ConfigureAwait(false);
+                return CreatedAtAction(
+                    nameof(GetAddressById),
+                    new
+                    {
+                        id = address.Id,
```

Comment:
why do we pass address.Id and address? 
address contains address.Id already

Reply:
> why do we pass address.Id and address?
> address contains address.Id already

otherwise, if we pass only address.id, it will not be working. I have tested it. Both id and address should be passed.